AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -84,15 +92,15 @@ public final class APIClient: APIClientProtocol, AsyncAPIClientProtocol {
         let result = try await urlSession
             .data(for: try buildUrlRequest(from: request)) as (data: Data, urlResponse: URLResponse)
         let httpResult = try URLSessionSuccess(data: result.data, response: result.urlResponse)
-        return try Self.handle(httpResult, request)
+        return try handle(httpResult, request)
     }
     
     /// :nodoc:
     public func perform<R: Request>(_ request: R, completionHandler: @escaping CompletionHandler<R.ResponseType>) {
         do {
             urlSession.dataTask(with: try buildUrlRequest(from: request)) { result in
                 let result = result
-                    .flatMap { response in .init(catching: { try Self.handle(response, request) }) }
+                    .flatMap { response in .init(catching: { try self.handle(response, request) }) }
```

Comment:
this is probably nothing, but can you make sure no retain cycle in here between `urlSession` and `self`, because I think maybe now `self.urlSession` retains self until the request is done.

probably the retain cycle is temporary for the duration of the request, but its worth checking that APIClient instance is properly deallocated when the request is done.

just make a request then when its done, make a break point in APIClient.deinit to make sure its called or something

Reply:
I have tested the functionality with following code:
```
class ViewController: UIViewController {
    
    var apiClient: APIClient? = APIClient(apiContext: APIContext())
    override func viewDidLoad() {
        super.viewDidLoad()
        let invalidGetRequest = GetUsersRequest(userId: "xxxxxx")
        apiClient?.perform(invalidGetRequest) { result in
            switch result {
            case .success:
                print("success")
            case let .failure(error):
                print("failure")
            }
        }
        self.apiClient = nil
    }
}
```
and saw that apiClient is not set to nil until session is finished. So, it is temporary. However, I think it is a better idea to check for self, still. What do you think?

Reply:
yeah, just make self weak I would say just to be safe :-)